x/mongo/driver/auth/creds/awscreds.go

```diff
@@ -8,181 +8,115 @@ package creds
 
 import (
 	"context"
-	"encoding/json"
 	"errors"
-	"io/ioutil"
-	"net/http"
 	"os"
 	"time"
 
-	"go.mongodb.org/mongo-driver/x/mongo/driver/auth/internal/awsv4"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/credentials/endpointcreds"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"go.mongodb.org/mongo-driver/internal/uuid"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
 )
 
-type ecsResponse struct {
-	AccessKeyID     string `json:"AccessKeyId"`
-	SecretAccessKey string `json:"SecretAccessKey"`
-	Token           string `json:"Token"`
-}
-
 const (
-	awsRelativeURI     = "http://169.254.170.2/"
-	awsEC2URI          = "http://169.254.169.254/"
-	awsEC2RolePath     = "latest/meta-data/iam/security-credentials/"
-	awsEC2TokenPath    = "latest/api/token"
-	defaultHTTPTimeout = 10 * time.Second
+	awsRelativeURI = "http://169.254.170.2/"
+	expiryWindow   = 5 * time.Minute
 )
 
-// AwsCredentials contains AWS credential fields.
-type AwsCredentials struct {
-	Username string
-	Password string
-	Token    string
-}
-
-// ValidateAndMakeCredentials validates credential fields and packs them into awsv4.StaticProvider.
-func (ac AwsCredentials) ValidateAndMakeCredentials() (*awsv4.StaticProvider, error) {
-	if ac.Username != "" && ac.Password == "" {
-		return nil, errors.New("ACCESS_KEY_ID is set, but SECRET_ACCESS_KEY is missing")
-	}
-	if ac.Username == "" && ac.Password != "" {
-		return nil, errors.New("SECRET_ACCESS_KEY is set, but ACCESS_KEY_ID is missing")
-	}
-	if ac.Username == "" && ac.Password == "" && ac.Token != "" {
-		return nil, errors.New("AWS_SESSION_TOKEN is set, but ACCESS_KEY_ID and SECRET_ACCESS_KEY are missing")
-	}
-	if ac.Username != "" || ac.Password != "" || ac.Token != "" {
-		return &awsv4.StaticProvider{Value: awsv4.Value{
-			AccessKeyID:     ac.Username,
-			SecretAccessKey: ac.Password,
-			SessionToken:    ac.Token,
-		}}, nil
-	}
-	return nil, nil
-}
-
-// AwsCredentialProvider provides AWS credentials.
+// AwsCredentialProvider wraps AWS credentials.
 type AwsCredentialProvider struct {
-	HTTPClient *http.Client
+	Value *credentials.Credentials
 }
 
-func executeAWSHTTPRequest(ctx context.Context, httpClient *http.Client, req *http.Request) ([]byte, error) {
-	ctx, cancel := context.WithTimeout(ctx, defaultHTTPTimeout)
-	defer cancel()
-	resp, err := httpClient.Do(req.WithContext(ctx))
-	if err != nil {
-		return nil, err
+// NewAwsCredentialProvider generates new AwsCredentialProvider
+func NewAwsCredentialProvider(v credentials.Value) (AwsCredentialProvider, error) {
```

Comment:
The Python implementation removes much of the logic: https://github.com/mongodb/pymongo-auth-aws/pull/13/files#diff-08503818a3e26253810d210133d82a4021995bc459ddd242bb3905e44fba1183L96

If we proceed with using the AWS SDK, can that also be done? This appears to work (:
```go
func NewAwsCredentialProvider(v credentials.Value) (AwsCredentialProvider, error) {
	creds := defaults.Get().Config.Credentials
	// TODO: configure expiration?
	return AwsCredentialProvider{creds}, nil
}
```